### PR TITLE
pillow: Fix build on non-NixOS systems

### LINKED ISSUE #36317: pillow fails to build on non-nixos system
## Issue description

I tried to build `python{2,3}Packages.pillow` on ArchLinux and it [failed](https://gist.github.com/timokau/22d9933b38a5bf1cf3381a8f0e593737) because of some problem with `libjpeg`.

The same build works under nixos.

### Steps to reproduce

Try to build `pillow` (either python version) on a non-nixos machine. Not sure if this is reliably reproducable.

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.23-1-lts, Arch Linux, noversion`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixpkgs-18.03pre129189.c2fbd472a4e"`
 - channels(timo): `"nixpkgs-18.03pre129189.c2fbd472a4e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

The build was from nixpkgs checkout 76a01ad9d.

### PR DESCRIPTION
**Pull request by andrew-d**
###### Motivation for this change
The Pillow install script will, by default, add paths like `/usr/lib` and
`/usr/include` to the search paths.  This can break things when building
on a non-NixOS system that has some libraries installed that are not
installed in Nix (for example, Arch Linux has jpeg2000 but Nix doesn't
build Pillow with this support).

We solve this by telling Pillow to knock it off 😄

Fixes #36317

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
---

cc @timokau

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg pythonPackages.pillow python3Packages.pillow

**Comment by timokau** [comment]
Wow, that was fast! Thanks

The build now succeeds.

It seems like this problem should have been prevented by sandboxing. I thought I have sandboxing enabled (`sandbox = true` in `~/nix.conf`) but is there a way to confirm nix is actually using sandboxing?

**Comment by Mic92** [comment]
nix 2.0 has a `nix show-config`, in older nix versions you can try to download something in the sandbox.

**Comment by timokau** [comment]
`nix show-config` tells me `sandboxing = true` while `nix-info` tells me its disabled. Apparently I need to be a trusted user to enable sandboxing? Seems backwards to me, trust should only be needed to disable it right? If thats the case, its also a special case of https://github.com/NixOS/nix/issues/495

**Comment by andrew-d** [comment]
@Mic92 - Forgive me if I'm mistaken, but did you forget to add the `build` prefix in your command to @GrahamcOfBorg?

**Comment by Mic92** [comment]
@GrahamcOfBorg build pythonPackages.pillow python3Packages.pillow

File: pkgs/development/python-modules/pillow/default.nix [hunk 1]
```diff
@@ -31,6 +31,9 @@ buildPythonPackage rec {
     ++ stdenv.lib.optionals (isPyPy) [ tk libX11 ];
 
   # NOTE: we use LCMS_ROOT as WEBP root since there is not other setting for webp.
+  # NOTE: we patch the `disable_platform_guessing` setting here, instead of
+  # passing the `--disable-platform-guessing` command-line option, since the
+  # command-line option doesn't work when we run tests.
```
Comment (Mic92) [inline pkgs/development/python-modules/pillow/default.nix]: Can you also summarize here briefly why, this has to be patched? Basically what you wrote in the commit message.

Comment (andrew-d) [reply pkgs/development/python-modules/pillow/default.nix]: @Mic92 - Done; amended and force-pushed the commit.

**Pull request merged**